### PR TITLE
Telegram Connection pool is full (and other tweaks)

### PR DESCRIPTION
**Pull request by Gobberwart**
## Short Description:

Increase connection pool size to eliminate connection pool is full errors. 
Removed unnecessary code from get_player_stats
Minor text tweaks (replace "spin up" with "start")

## Fixes/Resolves/Closes (please use correct syntax):
- #4977

### CODE REVIEW

**Comment by solderzzc** [comment]
Yes, the 'Connection pool is full' message is the only thing I can see on the console.

**Comment by Gobberwart** [comment]
Hmmm CI failed with error

```
======================================================================
FAIL: test_api_call_throttle_should_fail (tests.api_wrapper_test.TestApiWrapper)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/timeout_decorator/timeout_decorator.py", line 69, in new_function
    return function(*args, **kwargs)
  File "/home/travis/build/PokemonGoF/PokemonGo-Bot/tests/api_wrapper_test.py", line 116, in test_api_call_throttle_should_fail
    request.call()
AssertionError: TimeoutError not raised
```

You think that's related? Shouldn't be but not sure... Any thoughts?

**Comment by Gobberwart** [comment]
```
    @timeout(1) # expects a timeout
    def test_api_call_throttle_should_fail(self):
        request = FakeApi().create_request()
        request.is_response_valid = MagicMock(return_value=True)
        request.requests_per_seconds = 5

        with self.assertRaises(TimeoutError):
            for i in range(request.requests_pe

**Comment by Gobberwart** [comment]
Never mind, build passed now without me changing anything. Should be ready to merge if someone wants to approve it.

File: pokemongo_bot/event_handlers/telegram_handler.py [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
 DEBUG_ON = False
 
-class FileIOException(Exception):
-    pass
-
 class TelegramClass:
 
     update_id = None
-
+    session = None
```
Comment (sohje) [inline pokemongo_bot/event_handlers/telegram_handler.py]: What is this for?

File: pokemongo_bot/event_handlers/telegram_handler.py [hunk 2]
```diff
@@ -1,29 +1,25 @@
 # -*- coding: utf-8 -*-
 from pokemongo_bot.event_manager import EventHandler
 from pokemongo_bot.base_dir import _base_dir
```
Comment (sohje) [inline pokemongo_bot/event_handlers/telegram_handler.py]: dont need anymore

File: pokemongo_bot/event_handlers/telegram_handler.py [hunk 3]
```diff
@@ -1,29 +1,25 @@
...
     update_id = None
-
+    session = None
 
     def __init__(self, bot, master, pokemons, config):
         self.bot = bot
+        request.CON_POOL_SIZE = 16
         with self.bot.database as conn:
             # initialize the DB table if it does not exist yet
             initiator = TelegramDBInit(bot.database)
```
Comment (sohje) [inline pokemongo_bot/event_handlers/telegram_handler.py]: Does initiator var used here?

File: pokemongo_bot/event_handlers/telegram_handler.py [hunk 4]
```diff
@@ -1,29 +1,25 @@
...
 from pokemongo_bot.base_dir import _base_dir
-import json
-import os
 import time
 import telegram
 import thread
 import re
-from pokemongo_bot.datastore import Datastore
 import pprint
+from pokemongo_bot.datastore import Datastore
```
Comment (sohje) [inline pokemongo_bot/event_handlers/telegram_handler.py]: Do we really need datastore import?

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 3 above)
Comment (DBa2016) [inline pokemongo_bot/event_handlers/telegram_handler.py]: The initiator variable is not used anywhere, but I was not sure whether a class instance not assigned to a variable would at all lead to a constructor call.

And yes, I wanted it to be an extra class in order to be able to call this from TelegramClass and TelegramHandler without duplicating the code.

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 4 above)
Comment (DBa2016) [inline pokemongo_bot/event_handlers/telegram_handler.py]: I think it is needed for database operations, as already explained on the other PR where the same question was asked. You are welcome to remove it and test!

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 3 above)
Comment (sohje) [inline pokemongo_bot/event_handlers/telegram_handler.py]: Seems TelegramDBInit already initialized in TelegramHandler

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 4 above)
Comment (sohje) [inline pokemongo_bot/event_handlers/telegram_handler.py]: > (and other tweaks)

**Comment by Gobberwart** [comment]
@sohje Lots of comments on things that are not relevant to this PR. Yes, Telegram_handler.py needs a cleanup but not here.

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 3 above)
Comment (Gobberwart) [inline pokemongo_bot/event_handlers/telegram_handler.py]: @dba2016 I wasn't sure either, but it works :)

Comment (Gobberwart) [inline pokemongo_bot/event_handlers/telegram_handler.py]: @sohje not part of this PR, but it's in both classes due to the order in which they *might* be called. Both classes try to access db tables and will generate an error if tables don't exist.

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 4 above)
Comment (Gobberwart) [inline pokemongo_bot/event_handlers/telegram_handler.py]: I'll remove it and see what happens :)

**Comment by sohje** [comment]
@Gobberwart You already done some cleanup with this PR. That's why i commented. This PR about fix pool size, half cleanup also added?
```python
request.CON_POOL_SIZE = 16
```

**Comment by solderzzc** [comment]
👍

**Comment by solderzzc** [comment]
Let's remove the mass log with no much meaning on console.

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 4 above)
Comment (Gobberwart) [inline pokemongo_bot/event_handlers/telegram_handler.py]: Doesn't seem to make any difference. Needs further testing though (another PR)

File: pokemongo_bot/event_handlers/telegram_handler.py (hunk 1 above)
Comment (Gobberwart) [inline pokemongo_bot/event_handlers/telegram_handler.py]: Nothing at all. Was part of something I was testing, now unneccessary. Removing before merging.

Comment (Gobberwart) [inline pokemongo_bot/event_handlers/telegram_handler.py]: Removed in latest commit. Good catch, thanks.

**Comment by Gobberwart** [comment]
@sohje I only updated things that irritated me. In fact, this PR was originally going to be just about removing code from get_player_stats but I figured I'd look at connection pool while I was at it and that's probably more important.

**Comment by sohje** [comment]
@Gobberwart Okay, good. I dont mean any harm 👍

**Comment by Gobberwart** [comment]
@sohje All good mate, just don't want the PR bogged down in unrelated minutiae.

**Comment by Gobberwart** [comment]
OK to merge now then? Please approve.

**Comment by solderzzc** [comment]
👍   Don't pretend you can not merge :)

**Comment by Gobberwart** [comment]
LOL I know I can, I just prefer to let someone else check my work first :)
